realm/realm-library/src/androidTest/java/io/realm/QueryTests.java

```diff
@@ -80,6 +82,8 @@
 
     @Before
     public void setUp() throws Exception {
+        Realm.init(InstrumentationRegistry.getTargetContext());
```

Comment:
@cmelchior 
I tested it locally & it looks like `TestRealmConfigurationFactory#apply` is not called (which is responsible to call `Realm.init(InstrumentationRegistry.getTargetContext());`) when using `base` flavour. If you switch to `objectServer` flavour the rule is applied correctly.  

It looks related to the `io.realm.buildtransformer.RealmBuildTransformer`, I disabled the transformer & run the test again using the `base` flavour and it succeeded. It seems like we have a bug with the Transformer that needs investigation

Reply:
Nice catch. Thanks 👍